manticore/core/memory.py

```diff
@@ -771,13 +772,54 @@ def read(self, addr, size):
 
         return result
 
+
+    def start_write_trace(self):
+        '''
+        Begin recording all writes. Retrieve all writes with `stop_write_trace()`
+
+        :return: None
+        '''
+        print '[d] starting trace..'
+        self._recording_stack.append([])
+
+    def stop_write_trace(self):
+        '''
+        Stop recording trace and return a `list[(address, value)]` of all the writes
+        that occurred, where `value` is of type list[str]. Can be called without
+        intermediate `stop_write_trace()`.
+
+        For example,
```

Comment:
if we ever sphinx generate this, i don't think the code will output right, i think a `::`  will specify a code block  http://www.sphinx-doc.org/en/stable/rest.html#source-code